dataikuapi/dss/ml.py

```diff
@@ -315,43 +319,1016 @@ def save(self):
+        self._algo_settings[self.name] = value
+
+    def get_value(self):
+        """
+        :return: current value
+        :rtype: str
+        """
+        return self._algo_settings[self.name]
+
+    def get_all_possible_values(self):
+        """
+        :return: list of possible values for this hyperparameter
+        :rtype: list of str
+        """
+        return self.accepted_values
+
+
+class PredictionAlgorithmSettings(dict):
+    """
+    Object to read and modify the settings of a prediction ML algorithm.
+
+    Do not create this object directly, use :meth:`DSSMLTask.get_algorithm_settings` instead
+    """
+
+    def __init__(self, raw_settings, hyperparameter_search_params):
+        super(PredictionAlgorithmSettings, self).__init__(raw_settings)
+        self._hyperparameter_search_params = hyperparameter_search_params
+        self._hyperparameters_registry = dict()
+        self._attr_to_json_remapping = dict()
+
+    def __setattr__(self, attr_name, value):
+        if not hasattr(self, attr_name):
+            # call from __init__
+            super(PredictionAlgorithmSettings, self).__setattr__(attr_name, value)
+        else:
+            if attr_name in self._attr_to_json_remapping:
+                # attribute name and json key mismatch (e.g. "lambda", "alphaMode")
+                attr_name = self._attr_to_json_remapping[attr_name]
+            if attr_name in self._hyperparameters_registry:
+                # syntactic sugars
+                target = self._hyperparameters_registry[attr_name]
+                if isinstance(target, (SingleValueHyperparameterSettings, SingleCategoryHyperparameterSettings)):
+                    target.set_value(value)
+                elif isinstance(target, CategoricalHyperparameterSettings):
+                    target.set_values(value)
+                elif isinstance(target, NumericalHyperparameterSettings):
+                    raise Exception("Invalid assignment of a NumericalHyperparameterSettings object")
+                else:
+                    # simple parameter
+                    assert isinstance(value, type(target)), "Invalid type {} for parameter {}: expected {}".format(type(value), attr_name, type(target))
+                    super(PredictionAlgorithmSettings, self).__setattr__(attr_name, value)
+                    self[attr_name] = value
+                    self._hyperparameters_registry[attr_name] = value
+            else:
+                # other cases (properties setter, new attribute...)
+                super(PredictionAlgorithmSettings, self).__setattr__(attr_name, value)
+
+    def _maybe_register_attr_json_mismatch(self, json_key, attr_name):
+        if attr_name is not None:
+            self._attr_to_json_remapping[attr_name] = json_key
+
+    def _register_numerical_hyperparameter(self, json_key, attr_name=None):
+        self._maybe_register_attr_json_mismatch(json_key, attr_name)
```

Comment:
Might be a tiny bit more readable to rename method to `_register_attr_json_mismatch` and 
```suggestion
        if attr_name is not None:
            self._register_attr_json_mismatch(json_key, attr_name)
```

Reply:
Why not full inline instead of creating a method then ?
```suggestion
        if attr_name is not None:
            self._attr_to_json_remapping[attr_name] = json_key
```